tests/visualization_tests/matplotlib_tests/test_pareto_front.py

```diff
@@ -52,19 +54,36 @@ def test_plot_pareto_front_2d(
     study.enqueue_trial({"x": 0, "y": 1})
     study.optimize(lambda t: [t.suggest_int("x", 0, 1), t.suggest_int("y", 0, 1)], n_trials=3)
 
+    constraints_func: Optional[Callable[[FrozenTrial], Sequence[float]]]
+    if use_constraints_func:
+        # (x, y) = (1, 0) is infeasible; others are feasible.
+        def constraints_func(t: FrozenTrial) -> Sequence[float]:
+            return [1.0] if t.params["x"] == 1 and t.params["y"] == 0 else [-1.0]
+
+    else:
+        constraints_func = None
+
     figure = plot_pareto_front(
         study=study,
         include_dominated_trials=include_dominated_trials,
         axis_order=axis_order,
+        constraints_func=constraints_func,
         targets=targets,
     )
     assert len(figure.get_lines()) == 0
 
-    if axis_order is not None:
-        pareto_front_points = np.array([[1.0, 0.0], [0.0, 1.0]])[:, axis_order]
+    if constraints_func is not None:
+        if axis_order is not None:
+            pareto_front_points = np.array([[0.0, 1.0]])[:, axis_order]
+        else:
+            pareto_front_points = np.array([[0.0, 1.0]])
+        assert pareto_front_points.shape == (1, 2)
     else:
-        pareto_front_points = np.array([[1.0, 0.0], [0.0, 1.0]])
-    assert pareto_front_points.shape == (2, 2)
+        if axis_order is not None:
+            pareto_front_points = np.array([[1.0, 0.0], [0.0, 1.0]])[:, axis_order]
+        else:
+            pareto_front_points = np.array([[1.0, 0.0], [0.0, 1.0]])
+        assert pareto_front_points.shape == (2, 2)
```

Comment:
This test case seems to follow out-of-date code. In the current plotly tests, manually prepared data array is compared with data points in generated figures. Could you update the test code based on the current test cases?

https://github.com/optuna/optuna/blob/445048a74c9090e60a82a49605044cc42727642a/tests/visualization_tests/test_pareto_front.py#L90-L107